draft-ietf-quic-tls.md

```diff
@@ -1276,10 +1276,31 @@ If 0-RTT keys are available (see {{enable-0rtt}}), the lack of replay protection
 means that restrictions on their use are necessary to avoid replay attacks on
 the protocol.
 
-A client MUST only use 0-RTT keys to protect data that is idempotent.  A client
-MAY wish to apply additional restrictions on what data it sends prior to the
-completion of the TLS handshake.  A client otherwise treats 0-RTT keys as
-equivalent to 1-RTT keys, except that it MUST NOT send ACKs with 0-RTT keys.
+Of the frames defined in {{QUIC-TRANSPORT}}, the STREAM, RESET_STREAM, and
+CONNECTION_CLOSE frames are potentially unsafe for use with 0-RTT as they
+carry application data.  Application data that is received in 0-RTT could cause
+an application at the server to process the data multiple times rather than
+just once. Additional actions taken by a server as a result of processing
+replayed application data could have unwanted consequences. A client therefore
+MUST only use 0-RTT for application data that is permitted by the application
+that is in use.
+
+An application protocol that uses QUIC MUST include a profile that defines
+acceptable use of 0-RTT; otherwise, 0-RTT can only be used to carry QUIC frames
+that do not carry application data. For example, a profile for HTTP is
+described in {{?HTTP-REPLAY=RFC8740}}.
```

Comment:
Suggest rephrasing MUST since there is a way out in not using 0-RTT. For example, if an application is connecting over QUIC 0-RTT, the application protocol MUST ...

Reply:
I specifically chose this phrasing.  I know that prohibiting 0-RTT is a valid way to define a profile, but I think that is unnecessary for QUIC.  This would still allow that approach, but I would prefer a more nuanced profile that said "don't send application data in 0-RTT" than "don't send 0-RTT".